TwitterBlueBird/Controllers/TweetsController.cs

```diff
@@ -36,7 +36,47 @@ public ActionResult Details(int? id)
             return View(tweet);
         }
 
-        protected override void Dispose(bool disposing)
+		// Ajax POST: Tweets/RateAngry/:ID
+		[HttpPost]
+		public ActionResult RateHappy(int? id)
+		{
+			if (id != null)
+			{
+				Tweet tweet = db.Tweets.Find(id);
+				if (tweet != null && tweet.Mood == null)
+				{
+					tweet.Mood = "Happy";
+					using (var context = new TwitterAPIContainer())
+					{
+						context.Entry(tweet).State = EntityState.Modified;
+						context.SaveChanges();
+					}
+				}
+			}
+			return Content(id.ToString());
```

Comment:
If this method is called with a null id it will fail here, but this is not normally possible in this POST request; in the future (when flash messages are added) such rare failures will be handled more properly.